### DISCUSSION TITLE
Tesla preserve, how

### DISCUSSION THREAD

**Discussion by rob0101** [amberelectric/public-api#184] [Q&A] [answered]
Amber can put a powerwall in preserve mode.

I've yet to see a Tesla api endpoint to achieve this.

while we wait for the Amber api to enable this, how's it done?

**Comment by purcell-lab** [comment]
Amber use Preserve mode for a few different functions so there are a few different methods achieve the same outcome for Powerwall.

Load > Solar and you don't want battery to discharge (charging EV for example)
- set Powerwall backup reserve to match state of charge, battery SOC will remain fixed and additional power requirements will be drawn from the grid. If Solar > Load additional energy will be used to charge battery, which is often what I want but Amber preserve doesn't support.

Solar > Load but you want to export excess to Grid and not charge battery. (High feed in for example morning peak)
- switch to TBC hack (PEAK window), which will export / discharge battery+solar, switch allowed_exports endpoint to pv_only or never to preserve battery.

**Reply by rob0101** [reply]
Thanks Mark, but I'm looking for a real preserve rather than a faked one.

**Reply by rob0101** [reply]
Just re-read my question, and noticed the ambiguity. 

The question is how's it done by Amber ?  Are they using an endpoint that's actually available in the public api.

**Reply by purcell-lab** [reply]
From my understanding Amber don't use the consumer FleetAPI nor the Unofficial Owner API.

There is another uber secret API for VPP paying customers like Amber.

I don't fully understand the mechanics and everyone I speak to is under NDA.

I think it is related to autobidder/ powerhub on the Tesla end:
https://www.tesla.com/support/energy/tesla-software
https://www.tesla.com/support/energy/tesla-software/autobidder
https://www.tesla.com/support/energy/tesla-software/powerhub

**Reply by madpilot** [reply, marked as answer]
Basically what Mark said, though I t’s nothing to do with auto bidder.

We have an API which allows us to set a target power value. To preserve we set it to zero.

**Reply by rob0101** [reply]
"target power" as in,  positive to charge and negative to discharge at that rate (or vice-versa) ?

**Reply by madpilot** [reply]
Yep. I can’t remember the direction off the top off my head.

**Reply by rob0101** [reply]
Please Santa, can we have a target power in the Amber API.  We promise to be good.

**Reply by lhovo** [reply]
This is exactly what I would like to get my own automations working.

**Comment by damienkent** [comment]
> Please Santa, can we have a target power in the Amber API. We promise to be good.

Yes please! Even read only would be useful but manual control would be awesome.
